qutebrowser/browser/webengine/notification.py

```diff
@@ -361,13 +361,22 @@ def _on_adapter_clicked(self, notification_id: int) -> None:
             return
 
         try:
+            self._focus_first_matching_tab(notification)
             notification.click()
         except RuntimeError:
             # WORKAROUND for
             # https://www.riverbankcomputing.com/pipermail/pyqt/2020-May/042918.html
             log.misc.debug(f"Ignoring click request for notification {notification_id} "
                            "due to PyQt bug")
 
+    def _focus_first_matching_tab(self, notification: "QWebEngineNotification") -> None:
+        for win_id in objreg.window_registry:
+            tabbedbrowser = objreg.get("tabbed-browser", window=win_id, scope="window")
+            for idx, tab in enumerate(tabbedbrowser.widgets()):
+                if tab.url().matches(notification.origin(), QUrl.RemovePath):
```

Comment:
I assume the `QUrl.RemovePath` is really needed?

Reply:
Yeah, `origin()` only includes the scheme, host and port, so the URL in the tab needs to be cleaned up.